d2/src/main/java/com/linkedin/d2/discovery/event/PropertyEventBusImpl.java

```diff
@@ -199,7 +199,13 @@ public void innerRun()
         // an "initialize", but if the bus has previously seen that property, we will treat
         // it as an "add" so that the publisher change will be transparent to the clients.
         boolean doAdd = _properties.containsKey(prop);
-        _properties.put(prop, value);
+        if (!doAdd || (doAdd && value != null))
```

Comment:
This means there are changes to the way `publishInitialize` handles a `null` value. Can you update the documentation accordingly?

Reply:
what do u mean 
> Can you update the documentation accordingly?

I think I have added the explain below. Could u show me ?

Reply:
Updated